systemtests/bin/make-symbols-zip.py

```diff
@@ -0,0 +1,193 @@
+
+    return os.stat(filename).st_size
+
+
+@click.command()
+@click.option(
+    "--auth-token", required=True, help="Auth token for symbols.mozilla.org.",
+)
+@click.option(
+    "--start-page", default=1, type=int, help="Page of SYM files to start with.",
+)
+@click.option(
+    "--max-size",
+    default=10_000_000,
+    type=int,
+    help="Max size in bytes resulting ZIP file can't exceed.",
+)
+@click.argument("outputdir")
+@click.pass_context
+def make_symbols_zip(ctx, auth_token, start_page, max_size, outputdir):
+    """
+    Builds a zip file of SYM files recently uploaded to symbols.mozilla.org.
+
+    Note: This requires an auth token for symbols.mozilla.org to view files.
+
+    """
+    # Figure out the ZIP file name and final path
+    zip_filename = datetime.datetime.now().strftime("symbols_%Y%m%d_%H%M%S.zip")
+    zip_path = os.path.join(outputdir, zip_filename)
+
+    click.echo("Generating ZIP file %s ..." % zip_path)
+    with tempfile.TemporaryDirectory(prefix="symbols") as tmpdirname:
+        sym_dir = os.path.join(tmpdirname, "syms")
+        tmp_zip_path = os.path.join(tmpdirname, zip_filename)
+
+        sym_files_url = urljoin(SYMBOLS_URL, "/api/uploads/files/")
+        sym_files_generator = get_sym_files(auth_token, sym_files_url, start_page)
+        for sym_filename, sym_size in sym_files_generator:
+            if sym_filename.endswith(".0"):
+                # Skip these because there aren't SYM files for them.
+                continue
+
+            is_try = False
+
+            if os.path.exists(tmp_zip_path):
+                # See if the new zip file is too big; if it is, we're done!
+                zip_size = os.stat(tmp_zip_path).st_size
+                click.echo("size: %s, max_size: %s" % (zip_size, max_size))
+                if zip_size > max_size:
+                    # Handle weird case where the first zip file we built was
+                    # too big--just use that.
+                    if not os.path.exists(zip_path):
+                        shutil.copy(tmp_zip_path, zip_path)
+                    break
+
+                # This zip file isn't too big, so copy it over.
+                shutil.copy(tmp_zip_path, zip_path)
+
+            click.echo(
+                click.style(
+                    "Adding %s (%s) ..." % (sym_filename, sym_size), fg="yellow"
+                )
+            )
+
+            # Download SYM file into temporary directory
+            if sym_filename.startswith("try/"):
+                sym_filename = sym_filename[4:]
+                is_try = True
+
+            if sym_filename.startswith("v1/"):
+                sym_filename = sym_filename[3:]
+
+            url = urljoin(SYMBOLS_URL, sym_filename)
+            if is_try:
+                url = url + "?try"
+            sym_file = os.path.join(sym_dir, sym_filename)
+            download_sym_file(url, sym_file)
+
+            # Build the new zip file
+            build_zip_file(tmp_zip_path, sym_dir)
```

Comment:
This has a rough runtime thing where it creates a ZIP file every time it adds a SYM file. SYM files are text, so they compress pretty well. We could probably improve this by guessing the compression ratio and adding a bunch of SYM files until we're 75% of the way there, then adding-and-checking.

I didn't do that because it adds complexity and I don't see myself running this script very often. Given that, I'd rather it be simpler than more performant.